tests/cpp/core/RestServerTests.cpp

```diff
@@ -88,17 +92,27 @@ BOOST_AUTO_TEST_CASE(testServerReturnsSimpleContent)
     const auto url = QString("http://localhost:%1/test").arg(server.getPort());
     const auto response = sendHttpRequest(url);
 
-    BOOST_CHECK_EQUAL(response.toStdString(), "Hello World!");
+    BOOST_CHECK_EQUAL(response.first, "Hello World!");
+    BOOST_CHECK_EQUAL(response.second, int(QNetworkReply::NoError));
 }
 
 BOOST_AUTO_TEST_CASE(block_all)
 {
     RestServer server;
-    server.block(zeroeq::http::Method::GET);
     server.get().handleGET("test", [] { return "TEST"; });
+    server.get().handlePUT("bla/bar", [] { return true; });
 
-    const auto url = QString("http://localhost:%1/test").arg(server.getPort());
-    const auto response = sendHttpRequest(url);
+    server.block(zeroeq::http::Method::PUT);
+
+    const auto urlGET =
+        QString("http://localhost:%1/test").arg(server.getPort());
+    const auto urlPUT =
+        QString("http://localhost:%1/bla/bar").arg(server.getPort());
+
+    const auto responseGET = sendHttpRequest(urlGET);
+    const auto responsePUT = sendHttpRequest(urlPUT);
 
-    BOOST_CHECK_EQUAL(response.toStdString(), "Forbidden");
+    BOOST_CHECK_EQUAL(responseGET.second, int(QNetworkReply::NoError));
```

Comment:
check all the values: responseGET.first/second, responsePUT.first/second.
You should also validate that doing a PUT works before doing the block.